devito/ir/support/basic.py

```diff
@@ -426,7 +426,25 @@ def distance(self, other):
             else:
                 v = i - j
                 if v.is_Number and v.is_finite:
-                    return Vector(S.ImaginaryUnit)
+                    # If i and j are numbers, we append an (I) distance
+                    if i.is_Number and j.is_Number:
+                        return Vector(S.ImaginaryUnit)
+                    # If both i and j are not numbers, there may be dimension-dependent
+                    # dependencies so we append the distance
+                    else:
+                        ret.append(v)
```

Comment:
we might want to append Infinite here, otherwise I think we would have never ended up in this place, because we would have never left the first big loop at the top of the method over the iteration intervals

Reply:
The equation pair
```
eq0 = Eq(u[0,y], 1)
eq1 = Eq(u[1,1], u[0, y+1])
```
would resulting in the execution ending up at line 435

Also I'm confused to why adding an Infinite here would help with never reaching this part of the code